src/containers/audio-rewards-page/store/slice.ts

```diff
@@ -2,32 +2,83 @@ import { createSlice, PayloadAction } from '@reduxjs/toolkit'
 
 import { AppState } from 'store/types'
 
+import { UserChallenge, ChallengeRewardID } from '../types'
+
 export type TrendingRewardsModalType = 'tracks' | 'playlists' | 'underground'
+export type ChallengeRewardsModalType = ChallengeRewardID
+
+type UserChallengesPayload = {
+  userChallenges: UserChallenge[] | null
+}
 
 type RewardsUIState = {
+  loading: boolean
   trendingRewardsModalType: TrendingRewardsModalType
+  challengeRewardsModalType: ChallengeRewardsModalType
+  userChallenges: UserChallenge[]
```

Comment:
Could also store as a dict w/ key as id if you wanted - looks like it's always accessed by id anyways